### ISSUE TITLE
ExceptionInInitializerError in static block with BeforeAll present causes test to be ignored instead of FAILED

### ISSUE THREAD

**Issue by astubbs** [junit-team/junit5#2386]
ExceptionInInitializerError in static block causes test to be ignored instead of FAILED if BeforeAll present. Is there something special about BeforeAll that alters the nature of this error which makes this a feature, not a bug? IMO an error in anyway should fail a test.

Run the class test target:

```
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

public class BugSimple {

    static {
        if (true)
            throw new RuntimeException("Error in static block");
    }

    @BeforeAll
    static void beforeAllLogSomething() {
        //noop
    }

    @Test
    public void testNoop(){
        //noop
    }
}
```

Gets:
```
Test ignored.

java.lang.ExceptionInInitializerError
```

`[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 s - in snip.BugSimple`

The test will be ignored. It should instead be failed.

But, if you comment out the `@BeforeAll` annotation, and it will fail as expected, instead of being ignored.

Expected:
```[INFO] Running snip.BugSimple
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.002 s <<< FAILURE! - in snip.BugSimple
[ERROR] snip.BugSimple.testNoop  Time elapsed: 0 s  <<< ERROR!
java.lang.ExceptionInInitializerError
	at snip.BugSimple.<clinit>(BugSimple.java:10)
```

## Context

 - Used versions (Jupiter/Vintage/Platform):
<junit.version>5.7.0-M1</junit.version>
<artifactId>junit-jupiter-engine</artifactId>

 - Build Tool/IDE:
Maven and IntelliJ IDEA

P.s.,
Love you work.

**Comment by marcphilipp** [comment]
JUnit Platform reports it correctly:
```
Container Events:
	Event [type = STARTED, testDescriptor = JupiterEngineDescriptor: [engine:junit-jupiter], timestamp = 2020-08-18T19:40:06.190583Z, payload = null]
	Event [type = STARTED, testDescriptor = ClassTestDescriptor: [engine:junit-jupiter]/[class:BugSimple], timestamp = 2020-08-18T19:40:06.192832Z, payload = null]
	Event [type = FINISHED, testDescriptor = ClassTestDescriptor: [engine:junit-jupiter]/[class:BugSimple], timestamp = 2020-08-18T19:40:06.211125Z, payload = TestExecutionResult [status = FAILED, throwable = java.lang.ExceptionInInitializerError]]
	Event [type = FINISHED, testDescriptor = JupiterEngineDescriptor: [engine:junit-jupiter], timestamp = 2020-08-18T19:40:06.214912Z, payload = TestExecutionResult [status = SUCCESSFUL, throwable = null]]
```

This was apparently fixed in Surefire 3.0.0-M5: https://issues.apache.org/jira/browse/SUREFIRE-1741

**Comment by marcphilipp** [comment]
@astubbs Feel free to comment here should Surefire 3.0.0-M5 not resolve the issue for you.
